Intercept banner

gtk-grafana opened the pull request:
Fixes: https://github.com/grafana/explore-logs/issues/505

<img width="1728" alt="image" src="https://github.com/grafana/explore-logs/assets/109082771/d221d444-8c5d-4345-b806-4396b30fbc59">


@stevendungan @zizzpudding @matryer @JStickler Please review my draft copy and let me know if we want to add/remove anything.

We'll need a link to the video as well.

stevendungan commented:
@gtk-grafana 
* will the banner re-appear each session or will it remember state after it's been dismissed?
* is it pinned to the top of the page, or will it move with the user as they vertically scroll?

@JStickler @zizzpudding @matryer curious your opinions, i think we could shrink the screen real estate it takes and reword the content a bit but this directionally what we want and close.l

JStickler commented on src/Components/IndexScene/InterceptBanner.tsx:
```diff
@@ -0,0 +1,56 @@
+          <br />
+
+          <div>
+            We&rsquo;re looking for help in shaping the future of Logs Explore. If you have any issues or suggestions,
```
Logs Explore or Explore Logs?

matryer replied:
`Explore Logs` for now.

JStickler commented on src/Components/IndexScene/InterceptBanner.tsx:
```diff
@@ -0,0 +1,56 @@
+            </a>{' '}
+            or{' '}
+            <a
+              href="https://github.com/grafana/explore-logs/releases"
```
Would it be better to link to a CHANGELOG file?    * looks around *  Doesn't look like this repo has a changelog file (yet?)?

JStickler commented:
>i think we could shrink the screen real estate it takes and reword the content a bit

I'm OK with the size.  I'll take another look at the text in the morning when I'm fresh and can make good words again.  😛

gtk-grafana commented:
> * will the banner re-appear each session or will it remember state after it's been dismissed?
> * is it pinned to the top of the page, or will it move with the user as they vertically scroll?
> 
> @JStickler @zizzpudding @matryer curious your opinions, i think we could shrink the screen real estate it takes and reword the content a bit but this directionally what we want and close.l

Its removal is saved in local storage, will show until user closes it. That is, as long as you use the same device, you'll only see it once
Currently pinned to the top of the page, not fixed. I think it's annoying enough as is

stevendungan commented:
ok, i have no further strong opinions, this does the task well enough. i'm good.

matryer commented on src/Components/IndexScene/InterceptBanner.tsx:
```diff
@@ -0,0 +1,56 @@
+      {!props.interceptDismissed && (
+        <Alert
+          severity={'info'}
+          title={'Explore logs is under active development in a preview state.'}
```
maybe we can say,

"Welcome to Explore Logs (Preview) - we're in active development so please expect things to change.

Check out our video or dive into our Get Started docs for tips on how to use the app.

Let us know what we got right and what we got wrong by sending us your feedback."

Can we put a thumbnail of the video in there (it's fine if it pops up in a new window)?

matryer replied:
there will be a Get Started doc page: https://github.com/grafana/explore-logs/pull/456/files#diff-885722bddf5b70461db4e3b67474d96e8e0e3aafa781f7c3005cc7a38fa91f63

zizzpudding commented:
A couple of thoughts. 

- Could you add a "don't show me again" ? If someone dismisses it without reading first, will they be able to get back to it?  There are links to lots of good content here, it would be a shame to not have access to it.  Maybe we should also have this information available off of a button in the ui? 
- Does the Github link to a repository of all known issues or does it link to "open a new issue" 
- I like Mat's conversational approach to the message.  I'm not sure with the link to what we got right and what we got wrong.  
- "right" good feedback : slack channel or survey?
- "wrong" bad feedback : GH issue?

gtk-grafana commented:
@JStickler @matryer Joan and I just met, this is what we came up with:

<img width="799" alt="image" src="https://github.com/grafana/explore-logs/assets/109082771/6ae3e7b3-4707-4a1c-aaaa-e7b072b5e04a">
<img width="1728" alt="image" src="https://github.com/grafana/explore-logs/assets/109082771/23096951-8ddd-4e38-bfca-f2b2125ac327">


Not sure where we'd put the video thumbnail. I'd suggest punting for now.
I'll check with @stevendungan as to if we'll have a video to link to by next week, if not we'll remove that bit.

JStickler commented:
@gtk-grafana Looks good!

matryer commented:
@gtk-grafana love how simple that is.

matryer commented:
@gtk-grafana is this ready for review yet?

gtk-grafana commented:
@matryer just waiting on the link for "Check out our video" and "Get started doc"

JStickler commented:
@gtk-grafana  the link to the docs is  `https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/   `

gtk-grafana commented:
Last update: removed video callout, updated links
<img width="862" alt="image" src="https://github.com/grafana/explore-logs/assets/109082771/9c6b3c06-89df-4f9c-a624-783da2760ffb">

Welcome to Explore Logs (public preview) - we're in active development so please expect things to change.
Check out our [Get started doc](https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/), or see [recent changes](https://github.com/grafana/explore-logs/releases).
Help us shape the future of the app. [Send us feedback](https://forms.gle/1sYWCTPvD72T1dPH9) or engage with us on [GitHub](https://github.com/grafana/explore-logs/?tab=readme-ov-file#explore-logs).

svennergr commented on src/Components/IndexScene/LayoutScene.tsx:
```diff
@@ -1,26 +1,54 @@
     const styles = useStyles2(getStyles);
     return (
       <div className={styles.bodyContainer}>
+        <InterceptBanner
+          interceptDismissed={interceptDismissed}
+          onRemove={() => {
+            model.dismiss();
+          }}
+        />
```
I think we rather want to move the `interceptDismissed` logic out of the component, no? 

```suggestion
        {!interceptDismissed && (<InterceptBanner
          onRemove={() => {
            model.dismiss();
          }}
        />)}
```

gtk-grafana replied:
🫡

svennergr commented on src/Components/IndexScene/LayoutScene.tsx:
```diff
@@ -1,26 +1,55 @@
+  interceptDismissed: boolean;
+}
+
+const interceptBannerStorageKey = 'interceptBannerStorageKey';
```
Can we prefix this with `pluginJson.id`, like we do with other localstorage keys?
https://github.com/grafana/explore-logs/blob/bfd4682c33ecf987679358102cb222922fc92b0f/src/services/store.ts#L3-L4

gtk-grafana replied:
Done

svennergr commented on src/Components/IndexScene/InterceptBanner.tsx:
```diff
@@ -0,0 +1,43 @@
+      >
+        <div>
+          Check out our{' '}
+          <a
+            className="external-link"
+            href="https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/   "
+          >
+            Get started doc
+          </a>
+          , or see{' '}
+          <a className={'external-link'} href="https://github.com/grafana/explore-logs/releases">
+            recent changes
+          </a>
+          .<br />
+          Help us shape the future of the app.{' '}
+          <a className={'external-link'} href="https://forms.gle/1sYWCTPvD72T1dPH9">
+            Send us feedback
+          </a>{' '}
+          or engage with us on{' '}
+          <a
+            className={'external-link'}
+            href="https://github.com/grafana/explore-logs/?tab=readme-ov-file#explore-logs"
+          >
+            GitHub
+          </a>
```
Adding `_blank` target.

```suggestion
          <a
            className="external-link"
            target="_blank"
            rel="noreferrer noopener"
            href="https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/   "
          >
            Get started doc
          </a>
          , or see{' '}
          <a
            target="_blank"
            rel="noreferrer noopener"
            className="external-link"
            href="https://github.com/grafana/explore-logs/releases"
          >
            recent changes
          </a>
          .<br />
          Help us shape the future of the app.{' '}
          <a
            target="_blank"
            rel="noreferrer noopener"
            className="external-link"
            href="https://forms.gle/1sYWCTPvD72T1dPH9"
          >
            Send us feedback
          </a>{' '}
          or engage with us on{' '}
          <a
            className="external-link"
            target="_blank"
            rel="noreferrer noopener"
            href="https://github.com/grafana/explore-logs/?tab=readme-ov-file#explore-logs"
          >
            GitHub
          </a>
```

gtk-grafana replied:
And done

svennergr commented on src/Components/IndexScene/InterceptBanner.tsx:
```diff
@@ -24,21 +24,21 @@ export function InterceptBanner(props: { onRemove: () => void }) {
           , or see{' '}
           <a
             className={'external-link'}
```
I know this is a nit, but can we change this to just `"eternal-link"` please?

gtk-grafana replied:
Yup, agreed and fixed

svennergr commented on src/Components/IndexScene/InterceptBanner.tsx:
```diff
@@ -15,7 +15,7 @@ export function InterceptBanner(props: { onRemove: () => void }) {
             className="external-link"
-            target={'_blank'}
+            target="_blank"
             href="https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/   "
```
```suggestion
            href="https://grafana.com/docs/grafana-cloud/visualizations/simplified-exploration/logs/"
```

gtk-grafana replied:
Fixed

svennergr reviewed (commented):
Sorry, still some small nits.

svennergr reviewed (approved):
![Raise The Roof Office](https://media1.giphy.com/media/lMVNl6XxTvXgs/giphy.gif?cid=47028fa8sx12d824xrvy8zo8bq3w00f23hoj11gqqzus7q0f&ep=v1_gifs&rid=giphy.gif&ct=g)
